src/main/kotlin/uk/gov/dluhc/printapi/database/entity/AedPrintRequest.kt

```diff
@@ -0,0 +1,101 @@
+package uk.gov.dluhc.printapi.database.entity
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.GenericGenerator
+import org.hibernate.annotations.Type
+import org.springframework.data.annotation.LastModifiedBy
+import org.springframework.data.jpa.domain.support.AuditingEntityListener
+import uk.gov.dluhc.printapi.database.repository.UUIDCharType
+import uk.gov.dluhc.printapi.database.repository.UseExistingOrGenerateUUID
+import java.time.Instant
+import java.time.LocalDate
+import java.util.UUID
+import javax.persistence.CascadeType
+import javax.persistence.Entity
+import javax.persistence.EntityListeners
+import javax.persistence.FetchType
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.OneToMany
+import javax.persistence.PrePersist
+import javax.persistence.Table
+import javax.persistence.Version
+import javax.validation.constraints.NotNull
+import javax.validation.constraints.Size
+
+@Table
+@Entity
+@EntityListeners(AuditingEntityListener::class)
+class AedPrintRequest(
+    @Id
+    @Type(type = UUIDCharType)
+    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = UseExistingOrGenerateUUID.NAME)
+    var id: UUID? = null,
+
+    @field:NotNull
+    @field:Size(max = 7)
+    var electoralNumber: String? = null,
+
+    @field:NotNull
+    @field:Size(max = 255)
+    var aedTemplateFilename: String? = null,
+
+    @field:NotNull
+    var issueDate: LocalDate = LocalDate.now(),
+
+    @field:NotNull
+    var requestDateTime: Instant? = null,
+
+    @field:NotNull
+    @field:Size(max = 255)
+    var userId: String? = null,
+
+    @OneToMany(cascade = [CascadeType.ALL], orphanRemoval = true, fetch = FetchType.EAGER)
+    @JoinColumn(name = "print_request_id", nullable = false)
+    var statusHistory: MutableList<AedPrintRequestStatus> = mutableListOf(),
+
+    // managed using JPA pre-persist annotated method below
+    var dateCreated: Instant? = null,
```

Comment:
Out of interest, why are we using the pre-persist method and why do you have `@CreationTimestamp` on `AedPrintRequestStatus` instead?

Reply:
It was how it was defined in the `PrintRequest` that this is copied from and it was needed there as the created date is used for sorting and it needs to be populated earlier than the `@CreationTimestamp` is populated.  Although we can use the `requestDateTime` in this flow - will refactor 👍